pkg/controllers/management/auth/project_cluster/cluster_handler_test.go

```diff
@@ -0,0 +1,117 @@
+package project_cluster
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	apisv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	mgmtFakes "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/fakes"
+	"github.com/rancher/wrangler/v3/pkg/generic/fake"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestClusterLifecycleCreateProjectRespectsUserPrincipalName(t *testing.T) {
+	var project *apisv3.Project
+
+	ctrl := gomock.NewController(t)
+	projects := fake.NewMockClientInterface[*apisv3.Project, *apisv3.ProjectList](ctrl)
+	projects.EXPECT().List(gomock.Any(), gomock.Any()).Return(&apisv3.ProjectList{}, nil).Times(1)
+	projects.EXPECT().Create(gomock.Any()).DoAndReturn(func(p *apisv3.Project) (*apisv3.Project, error) {
```

Comment:
nit: Can we get rid of `gomock.Any()` here, and replace it with the expected parameters?

Reply:
I don't think it's worth it, especially for the `Create` call which takes the whole `Project` object.

Reply:
I see the value in doing it. You are not verifying the right `Project` object (with the expected values) is created. You just verify `projects.Create()` is called. This could be improved by removing `gomock.any()`

However, this is a small improvement and I don't see this as a blocker. So feel free to ignore/resolve this.

Reply:
I probably should've bean more clear. There are two distinct ways to assert that the "right" project object is created:
- by offloading it to a Create call expectation
- by capturing the created object and explicitly asserting that parts or the whole object is correct
This particular test uses the latter. I don't see much value in the former because it makes the test more brittle.

Reply:
I agree though it can be useful in the case of simple type arguments. E.g. it can be used with the `List` expectation.